aws/s3_test.go

```diff
@@ -132,332 +161,331 @@ func (b TestS3Bucket) addObject(s SetupInfo, fileName string, fileBody string) e
-	if !shouldNuke {
-		if delCount != 0 {
+	// If we should not nuke the bucket then deleted bucket count should be 0
+	if !args.shouldNuke {
+		if delCount > 0 {
 			assert.Failf(t, "Should not nuke but got delCount > 0", "delCount: %d", delCount)
 		}
+		logging.Logger.Debugf("SUCCESS: Did not nuke bucket - %s", bucket.name)
 		return
 	}
 
 	// Verify that - after nuking test bucket - it should not exist
-	bucketNamesPerRegion, err := getAllS3Buckets(awsSession, time.Now().Add(1*time.Hour), []string{s.region}, s.bucketName, 100)
+	bucketNamesPerRegion, err := getAllS3Buckets(awsSession, time.Now().Add(1*time.Hour), []string{awsParams.region}, bucket.name, 100)
 	if err != nil {
 		assert.Failf(t, "Failed to list S3 Buckets", errors.WithStackTrace(err).Error())
 	} else {
-		assert.NotContains(t, bucketNamesPerRegion[s.region], aws.String(s.bucketName))
+		assert.NotContains(t, bucketNamesPerRegion[awsParams.region], aws.String(bucket.name))
+		logging.Logger.Debugf("SUCCESS: Nuked bucket - %s", bucket.name)
 	}
 }
 
-func TestNuke_S3Bucket_WithoutVersioning_AllObjects(t *testing.T) {
-	testNukeS3BucketWrapper(t, &TestNukeS3BucketArgs{
-		isVersioned:       false,
-		checkDeleteMarker: false,
-		objectCount:       10,
-		objectBatchsize:   1000,
-	}, true)
-}
+// TestNukeS3Bucket tests S3 bucket deletion
+func TestNukeS3Bucket(t *testing.T) {
+	t.Parallel()
 
-func TestNuke_S3Bucket_WithoutVersioning_BatchObjects(t *testing.T) {
-	testNukeS3BucketWrapper(t, &TestNukeS3BucketArgs{
-		isVersioned:       false,
-		checkDeleteMarker: false,
-		objectCount:       10,
-		objectBatchsize:   2,
-	}, true)
-}
+	type testCaseStruct struct {
+		name string
+		args TestNukeS3BucketArgs
+	}
 
-func TestNuke_S3Bucket_WithoutVersioning_BatchObjects_InvalidBatchSize(t *testing.T) {
-	testNukeS3BucketWrapper(t, &TestNukeS3BucketArgs{
-		isVersioned:       false,
-		checkDeleteMarker: false,
-		objectCount:       2,
-		objectBatchsize:   1001,
-	}, false)
-
-	testNukeS3BucketWrapper(t, &TestNukeS3BucketArgs{
-		isVersioned:       false,
-		checkDeleteMarker: false,
-		objectCount:       2,
-		objectBatchsize:   0,
-	}, false)
-}
+	var allTestCases []testCaseStruct
 
-func TestNuke_S3Bucket_WithVersioning_AllObjects(t *testing.T) {
-	testNukeS3BucketWrapper(t, &TestNukeS3BucketArgs{
-		isVersioned:       true,
-		checkDeleteMarker: false,
-		objectCount:       10,
-		objectBatchsize:   1000,
-	}, true)
-}
+	allTestCases = append(allTestCases, testCaseStruct{
+		"EmptyBucket",
+		TestNukeS3BucketArgs{isVersioned: false, checkDeleteMarker: false, objectCount: 0, objectBatchsize: 0, shouldNuke: true},
+	})
 
-func TestNuke_S3Bucket_WithVersioning_BatchObjects(t *testing.T) {
-	testNukeS3BucketWrapper(t, &TestNukeS3BucketArgs{
-		isVersioned:       true,
-		checkDeleteMarker: false,
-		objectCount:       10,
-		objectBatchsize:   3,
-	}, true)
-}
+	for _, bucketType := range []string{"NoVersioning", "Versioning"} {
+		isVersioned := bucketType == "Versioning"
```

Comment:
NIT: use `true` and `false` directly in the range var.

Reply:
Cannot do so as the tests need the Versioning v/s NonVersioning prefix 

```
--- PASS: TestNukeS3Bucket/NoVersioning_EmptyBucket (9.39s)
    --- PASS: TestNukeS3Bucket/NoVersioning_BatchObjects_InvalidBatchSize_Over (9.97s)
    --- PASS: TestNukeS3Bucket/Versioning_EmptyBucket (9.60s)
    --- PASS: TestNukeS3Bucket/Versioning_AllObjects (24.06s)
   ```

Having true/false in range will lead to reinterpreting true false and setting Versioning/nonVersioning

Reply:
You could do:

```go
fmt.Sprintf("Versioning_%v_EmptyBucket", isVersioning)
```

But it's not a big deal so can keep this as is.